src/components/Input.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { ChangeEvent, FC } from 'react'
+
+export interface InputProps {
+  type: string
+  placeholder: string
+  value: string
+  key?: string
+  onBlur?: (e: ChangeEvent<HTMLInputElement>) => void
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void
+  onFocus?: (e: ChangeEvent<HTMLInputElement>) => void
+}
+/**
+ * Input box component.
+ */
+const Input: FC<InputProps> = ({ key, type, value, placeholder, onBlur, onChange, onFocus }) => (
+  <input
+    key={key}
+    type={type}
+    placeholder={placeholder}
+    contentEditable={false}
+    value={value}
+    onBlur={onBlur}
+    onFocus={onFocus}
+    onChange={onChange}
+  />
+)
```

Comment:
Having a separate component makes it easier to reuse in ModalAuth as well as ModalInvites for now and any other place we need input field in future.

Reply:
I don't see any special reusable logic being used in this `Input`. It's basically the same interface you would d use with normal `input`. The only thing I noticed is `contentEditable` set to `false` which doesn't make any difference because the input is always editable unless made `readonly` or `disabled`. Please let me know if I am missing something here.

Reply:
Components should be separated if they have some substantial reuseable parts that will be needed in multiple places. I don't see any special reusable logic being used in this Input. It's basically the same interface you would use with normal input. The only thing I noticed is `contentEditable` set to false which doesn't make any difference because the input is always editable unless made `readonly` or `disabled`. Please let me know if I am missing something here.

Reply:
If I am understanding it correct, if I need the same logic in multiple components, I should duplicate the code in all places?

Reply:
I don't see any reusable logic here. The custom `Input` component has same interface as normal `input`. Can you explain me what is reusable here?

Reply:
If I search for `InputProps`, I see is defined in 3 different file - ModalAuth, ModalSignup and Input. I am not sure if that is the right way to have same defined in multiple spots. Similarly, if I search for `Input:`, I see it is also defined in same 3 files. My end goal is combine all and have the `interface InputProps` as well as `Input` defined just once and reused in every other place. 

That was my reason behind creating a separate component. Also refactoring other files, I haven't done as part of this change. That will be part of next one where the default screen will be login instead of welcome. 

If you still think it doesn't make sense, I will define it the file I am using `Input` component in. Please let me know.

Reply:
I don't quite understand here. There is no need for defining a separate `InputProps`, the default `input` already has all the necessary type definitions. For example, if you replace every custom `Input` with default `input`, it should work just fine.